cpp/src/strings/regex/config.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "regex.cuh"
```

Comment:
Include a `.cuh` into a `.cpp` file, this means that `.cuh` file doesn't contain device code, does it? If so, we may also need to rename it to `regex.hpp`.

Reply:
A `.cuh` file (like the `regex.cuh` one here) can contain both device code and host code. An `.hpp` should not contain any device code at all. The class used in the `regex.cuh` by this source file (`confg.cpp`) has host and device (and host-device) member functions. The code in this file only uses the host or host-device functions. Since this file (`config.cpp`) does not make any device calls it is named with the `.cpp` extension.